src/middlewares/trip.middleware.js

```diff
@@ -0,0 +1,16 @@
+import ResponseService from '../services/response.service';
+import RequestService from '../services/request.service';
+
+const tripMiddleware = {
+  checkIfTripRequestExist: async (req, res, next) => {
+    const { tripId } = req.params;
+    const findTrip = await RequestService.findRequestByProperty({ tripId });
```

Comment:
Also, check that the request belongs to the requester here

Reply:
To check if request belongs to requester has been done in `trip.controller.js`   `line 130`

Reply:
The check should be done in the middleware instead before it gets to the controller. The `findTripRequest` in `trip.controller.js#130` is unnecessary and redundant.